library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -370,9 +371,34 @@ private boolean updateLocked() {
             alarmManager.set(AlarmManager.RTC_WAKEUP, now + nextRetryTimeMillis, PendingIntent.getBroadcast(this, 0, intent, PendingIntent.FLAG_ONE_SHOT));
         }
 
+        if (!isActive) {
+            moveSubmittedTasksToBatchStatusIfNecessary();
+        }
+
         return isActive;
     }
 
+    private void moveSubmittedTasksToBatchStatusIfNecessary() {
+        List<FileDownloadInfo> allDownloads = downloadsRepository.getAllDownloads();
+        List<DownloadBatch> downloadBatches = batchRepository.retrieveBatchesFor(allDownloads);
```

Comment:
following @amlcurran 's comment, we could avoid doing this if we could `getAllBatches()` instead of getting all the downloads and then all the batches from those downloads (which I assume it's all the batches, correct me if this is wrong)